server/routes/errors.js

```diff
@@ -1,7 +1,52 @@
 'use strict'
 
+const fs = require('fs')
 const log = require('../logger')
-const {stringTemplate} = require('../utils')
+const {stringTemplate, assetDataURI} = require('../utils')
+
+let assetCache
+
+// Because asset requests are authenticated, we inline the images and CSS
+// that we need to serve to logged-out users, e.g. on the auth error page.
+function loadInlineAssets() {
+  if (assetCache) return new Promise((resolve) => resolve(assetCache))
+
+  const assets = {}
+  const assetPromises = ['branding.icon', 'branding.favicon'].map((key) => {
+    // Load essential images as base64 data-URIs
+    return new Promise((resolve, reject) => {
+      assetDataURI(stringTemplate(key)).then((img) => {
+        assets[key] = img
+        resolve()
+      }).catch((err) => {
+        // It's okay for users to have deleted these keys,
+        // but other errors should throw
+        if (err.code === 'ENOENT') { resolve(err) } else { throw err }
+      })
+    })
+  }).concat(
+    // Load the core stylesheet
+    new Promise((resolve, reject) => {
+      fs.readFile('public/errors.css', (err, data) => {
+        if (err) throw err
+        assets.css = data.toString()
```

Comment:
@isaacwhite if we still want to parse `background-image: url(/etc.png)` from the CSS and inline the images, I think this is where I would add it. I didn't end up needing any background images in this implementation, so I skipped this step for now.

Reply:
That makes sense, @chrisfrank. What do you think about leaving an inline comment for future PRs here in case someone else wants to add support for that later?